### PR TITLE
chore(ci): add github action to bump version & publish package

### LINKED ISSUE #859: Automate release of kaoto-ui to npm-js
### Please describe the feature that you want to propose

When a tag is created, I would like that a job is triggered to publish on npmjs

### PR DESCRIPTION
**Pull request by kahboom**
## Changes
- adds a new github action with steps to bump the version in `package.json` and subsequently publish the package to npm
- bump minimum node version to >= v16.x
- updates some emojis 🤪

implements #859

### CODE REVIEW

**Review by apupier — COMMENTED** [review]
I think that a requirement bump from node 14 to node 16 would require a 0.x to 0.y bump version but well as we are still in 0.x this can be fine.

File: .github/workflows/publish-npm-package.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: 'Checkout source code'
+        uses: actions/checkout@v3
+      # Setup .npmrc file to publish to npm
+      - name: 'cat package.json'
+        run: cat ./package.json
+      - name: 'Bump package.json version'
+        uses: 'phips28/gh-action-bump-version@master'
```
[the remark this answers is not in the document]
Comment (kahboom) [reply .github/workflows/publish-npm-package.yml:15]: sure, this was just from the examples they provided and use even for their own GitHub workflow. if we use a tag can I just do `phips28/gh-action-bump-version@v9.0.42`? if the tag is `v9.0.42`? or does that only work with branches?

[the remark this answers is not in the document]
Comment (Delawen) [reply .github/workflows/publish-npm-package.yml:15]: Each of our releases have a semantic tag: https://semver.org/

So this github action should be tied to a tag creation. Either by being triggered on tag creation or by creating the tag itself.

[the remark this answers is not in the document]
Comment (Delawen) [reply .github/workflows/publish-npm-package.yml:15]: I see it is run when a release is created. Yes, the checkout must checkout the tag associated to the release.

**Comment by kahboom** [comment]
@Delawen okay so i removed the trigger for it to run automatically on release, based on your suggestion

File: .github/workflows/publish-npm-package.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
+name: 🏷 Bump Version & 📦 Publish Package to npm
+jobs:
```
Comment (Delawen) [inline .github/workflows/publish-npm-package.yml:2]: We should have a 

`on: workflow_dispatch`

to be able to trigger it manually

Comment (kahboom) [reply .github/workflows/publish-npm-package.yml:2]: @Delawen can't you just go to the Actions tab and click "Run workflow"? where would you want to trigger it from?

**Review by mmelko — DISMISSED** [review]

**Review by mmelko — APPROVED** [review]
LGTM

**Pull request merged**